### PR TITLE
Redux - Issue 33 - Fix for the downcase and upcase malfunction

### PR DESCRIPTION
**Pull request by guyboertje**
Added a specific multi-stage test
Only include bug fix
TODO: refactor other methods

See this Closed PR for full discussion on original PR https://github.com/logstash-plugins/logstash-filter-mutate/pull/42

### CODE REVIEW

File: spec/filters/integration/multi_stage_spec.rb [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+  describe 'MUTATE-33: multi stage with json, grok and mutate, Case mutation' do
+    let(:event) do
+      "{\"message\":\"hello WORLD\",\"examplefield\":\"PPQQRRSS\"}"
+    end
+
+    let(:config) do
+      <<-CONFIG
+filter {
+    grok {
+        match => { "message" => "(?:hello|world) %{WORD:bar}" }
```
Comment (jordansissel) [inline spec/filters/integration/multi_stage_spec.rb]: similar comment to before - do we need grok here? I'm ok with this as an integration test.

If we keep grok, what is the `|world` part of the regexp for?

File: spec/filters/mutate_spec.rb [hunk 2]
```diff
@@ -70,18 +70,18 @@
...
-      insist { subject["floatme"] } == [1234.455]
-      reject { subject }.include?("rename1")
-      insist { subject["rename2"] } == [ "hello world" ]
-      reject { subject }.include?("removeme")
-
-      insist { subject }.include?("newfield")
-      insist { subject["newfield"] } == "newnew"
-      reject { subject }.include?("nosuchfield")
-      insist { subject["updateme"] } == "updated"
+      expect(subject["lowerme"]).to eq ['example']
```
Comment (jordansissel) [inline spec/filters/mutate_spec.rb]: Thank you for upgrading this from insist to rspec3-style expect!

**Comment by jordansissel** [comment]
Code and Tests looking good.

Can you update your commit message where the specs are converted from `insist` to `expect` that you made this change?

File: spec/filters/integration/multi_stage_spec.rb (hunk 1 above)
Comment (guyboertje) [inline spec/filters/integration/multi_stage_spec.rb]: I will remove the `|world`

**Comment by guyboertje** [comment]
OK on the commit message bit.

**Comment by jordansissel** [comment]
LGTM!

**Pull request merged**